pkg/pilot/cassandra/v3/pilot.go

```diff
@@ -80,9 +82,39 @@ func (p *Pilot) CmdFunc(pilot *v1alpha1.Pilot) (*exec.Cmd, error) {
 }
 
 func (p *Pilot) syncFunc(pilot *v1alpha1.Pilot) error {
-	if p.genericPilot.IsThisPilot(pilot) {
+	if !p.genericPilot.IsThisPilot(pilot) {
+		return nil
 	}
-	return nil
+	pilot, err := p.pilotLister.Pilots(pilot.Namespace).Get(pilot.Name)
+	if err != nil {
+		return err
+	}
```

Comment:
Don't need to obtain a copy of `pilot` - you already have one as the arg to the sync func which was just retrieved from the lister (in the callsite for this function)